tests/drivers/uart/uart_async_api/testcase.yaml

```diff
@@ -1,3 +1,8 @@
+common:
+  platform_exclude: seeeduino_xiao serpente arduino_nano_33_iot atsamr21_xpro
```

Comment:
i thought we were trying to move away from this. How did this test started failing all of the sudden?

Reply:
@nashif the change to use DEVICE_DT_GET changed a runtime error into a compile error.  So these tests would build before but would never run correctly.

One issue is that if we enable UART ASYNC support in Kconfig, than its assumed ALL uart instances support that which may not be the case.

Reply:
I'm open to a better solution.